node/directives/checkUserAccess.ts

```diff
@@ -13,47 +70,7 @@ export class CheckUserAccess extends SchemaDirectiveVisitor {
       context: Context,
       info: any
     ) => {
-      const {
-        vtex: { adminUserAuthToken, storeUserAuthToken, logger },
-        clients: { identity, vtexId },
-      } = context
-
-      if (!adminUserAuthToken && !storeUserAuthToken) {
-        throw new AuthenticationError('No admin or store token was provided')
-      }
-
-      if (adminUserAuthToken) {
-        try {
-          await identity.validateToken({ token: adminUserAuthToken })
-        } catch (err) {
-          logger.warn({
-            error: err,
-            message: 'CheckUserAccess: Invalid admin token',
-            token: adminUserAuthToken,
-          })
-          throw new ForbiddenError('Unauthorized Access')
-        }
-      } else if (storeUserAuthToken) {
-        let authUser = null
-
-        try {
-          authUser = await vtexId.getAuthenticatedUser(storeUserAuthToken)
-          if (!authUser?.user) {
-            authUser = null
-          }
-        } catch (err) {
-          logger.warn({
-            error: err,
-            message: 'CheckUserAccess: Invalid store user token',
-            token: adminUserAuthToken,
-          })
-          authUser = null
-        }
-
-        if (!authUser) {
-          throw new ForbiddenError('Unauthorized Access')
-        }
-      }
+      await checkUserOrAdminTokenAccess(context, field.astNode?.name?.value)
```

Comment:
I think I'm missing something, here, but from what I understood we now have both directives:

1. `@auditAccess`. Always checks the token and throws errors when it's not present or valid.
2. `@checkAccessWithFeatureFlag`. Does the same as the other one, but only when the feature flag is enabled.

Why have both? The `auditAccess` will already do the check and throw before `checkAccessWithFeatureFlag` ever runs right? Shouldn't we only have `checkAccessWithFeatureFlag`, so that we can control when the check is made? And having both will make us call vtexid/identity twice when the feature flag is enabled and the check passes, since we'll check the same thing that has been checked by the first directive.

I think I misunderstood something though, can you explain the idea of the two directives with more details?

Reply:
I think you said about the `@checkUserAccess` instead of `@auditAccess`, is it?

Reply:
`@checkUserAccess` - Validates the storeUserToken or adminToken, but is not applied for all operations.
`@auditAccess` - Generate metrics about the use of the operation and if it uses the token or not.
`@checkAccessWithFeatureFlag` - It was created for temporary use. We put in some operations with `@auditAccess`, without token validation, and mapped in the [task](https://vtex-dev.atlassian.net/browse/B2BTEAM-1287). The intention is to control whether we can enable or disable token validation for these operations in production.

Reply:
Final question then: will we be able to remove this `auditAccess` directive after all the missing access checks are  fully launched? We won't need those metrics anymore when that happens right?

Reply:
Yes, this is the intention. I almost removed it in this PR, but I let it go for the next one.